### DISCUSSION TITLE
Close all apps before launching App Layouts

### DISCUSSION THREAD

**Discussion by pilniks** [rxhanson/RectanglePro-Community#556] [Ideas]
Hello,

I have a straightforward proposal that could significantly enhance the functionality of the application. By adding a feature to close all open apps in "App layouts" before launching a new layout, users could switch work modes quickly without leaving unused apps running in the background. Implementing a simple shortcut to close these apps would be a major improvement.

This feature would streamline the user experience, making it more efficient and user-friendly.

**Comment by rxhanson** [comment]
Thanks for the feedback. I have a similar item on my backlog already, although the idea there was for hiding the existing windows rather than quitting apps. I'll have to give this some thought.

**Comment by pilniks** [comment]
No problem. I thought closing apps would help to switch between for example private and work environment and save PCs resources at the same time.

**Comment by pilniks** [comment]
I just thought that this could be a killer feature for a fairly large group of people who have 8GB of RAM. Thanks to this, you could switch between environments, like for work, planning, or entertainment, without delays caused by memory slowdown. For sure I would like to have it on my 8GB Mac.

**Comment by BrentEubanks** [comment]
Just bought the app and came here to see if there was a feature request for exactly this.

I'd like to second a request for this functionality.  I have a saved layout "meeting prep" that I fire before I take a meeting, so I have all my notes, video call window, etc all set up.  I usually try to close all my existing windows first but sometimes I forget.  It would be nice to have that step automated as part of the setup, just to eliminate clutter.

FWIW, I'd like to have the options to either close or minimze running apps.  I find that my M2 Air doesn't seem to mind holding quite a few apps running in the background, so I'd probably generally opt to minimze.

Any chance that this feature will make it off the backlog ?

**Reply by rxhanson** [reply]
Thanks for purchasing. I didn't quite get to the point where I was confident adding this to the UI, but if you don't mind trying it out and letting me know how it works for you, I'd appreciate it. Enable it by executing the following Terminal command and restarting the app:

```
defaults write com.knollsoft.Hookshot appLayoutsHideOtherApps -int 1
```

**Reply by BrentEubanks** [reply]
Cool, thanks!

I just tried it and so far it seems to work as expected.  I'll live with it a while and give you more feedback later.

One thought is that it would be nice to have the option to minimize rather than hide the other windows.  Is that a feature you could add?

**Reply by rxhanson** [reply]
Although I think unminimizing programmatically might be possible it might not be something that works consistently in all scenarios with what's available as a 3rd party. I'd have to play around with it a bit more.

**Reply by rxhanson** [reply]
Also, thanks for trying this out!

**Reply by BrentEubanks** [reply]
Sorry I'm not following you.  Why would you need "unminimizing programmatically"?  My suggestion was to minimize the extra windows as an alternative to hiding them.  Unminimizing would be done by hand, afterward.

Although, now that you mention it, having the option to _reverse_ any layout operation would be pretty cool.  So when I'm done with my meeting, I can restore the state of things to the way they were before (even though that might not have been a saved layout).
But that's a whole other thing, I would think :).

**Reply by rxhanson** [reply]
Perhaps I was thinking ahead a bit on this without bringing you along :)

If you're good with having to unminimize manually, then I can think farther down that path, too. Initially I was thinking that this wouldn't be the most desirable option.

**Reply by BrentEubanks** [reply]
Well, to be honest I'm making up the workflow as I go.  I've never had this much screen real estate before and it implicates a different working style.  I'm also moving over from being primarily a windows user so figuring out new patterns as I go.

I think that ultimately having the option to restore would be pretty awesome, but it is a useful feature without that.  And having the restore function be optional/selectable is good, because sometimes the last thing you want is all the clutter back.

**Reply by BrentEubanks** [reply]
> Thanks for purchasing. I didn't quite get to the point where I was confident adding this to the UI, but if you don't mind trying it out and letting me know how it works for you, I'd appreciate it. Enable it by executing the following Terminal command and restarting the app:
> 
> ```
> defaults write com.knollsoft.Hookshot appLayoutsHideOtherApps -int 1
> ```

Hey Ryan

I've been using Rectangle with this setting turned on for a couple of days, and I wanted to give you some feedback.

As noted, it would be nice to have the option to either minimize open windows or hide the open apps.

In addition, it would be **very** helpful to have the ability to set this option on a per layout basis.

For example, I have a two-part workflow to set up for a meeting:

- Step 1: Activate Layout A, which opens my email, calendar, notetaking software, and browser and positions them on the screen.  I have Layout A set to open/launch any apps that are closed or hidden.  I also want Layout A to close/minimize/hide all other open windows.
- Step 2: From my calendar, I launch the video call app that is associated with the particular meeting that is coming up.  I bounce between three or four different meeting apps, so which app I launch at this point varies. 
- Step 3: Activate Layout B, which triggers each of my possible video call apps and tries to put each of them into the same drop target window.  With Layout B, I have NOT set the option to open/launch apps which are closed and hidden - the result is that whichever app I opened in step 2 gets dropped into the appropriate spot on my screen (which is an open spot left from Layout A), and ideally that's all that happens at this step.  For Step 3, I do NOT want Layout B to close/minimize/hide other open apps, because I don't want to disrupt the window layout that I established in Step 1.


Sorry if this is kind of a long, drawn-out explanation, but hopefully it makes sense and explains why I want this as a selectable option.

Is that something that seems feasible to you?

**Reply by rxhanson** [reply]
Thanks for trying this out, and for the feedback! That makes sense to me. I had been thinking about this differently but I like this and will likely roll it in this way when it makes it into the UI.

**Reply by BrentEubanks** [reply]
Cool.  Let me know if you have a beta you'd like me to try.  I'm eager to have this feature since I really want to set up the workflow I described and I can't quite do that with the app in its current state, so definitely keep me posted!
